packages/cli/src/index.ts

```diff
@@ -493,8 +470,46 @@ yargs
           choices: ['warn', 'error', 'off'] as ReadonlyArray<RuleSeverity>,
           default: 'warn' as RuleSeverity,
         },
+        format: {
+          hidden: true,
+          deprecated: true,
+        },
+        lint: {
+          hidden: true,
+          deprecated: true,
+        },
+        'skip-rule': {
+          hidden: true,
+          deprecated: true,
+          array: true,
+          type: 'string',
+        },
+        'max-problems': {
+          hidden: true,
+          deprecated: true,
+        },
       }),
     (argv) => {
+      const DEPRECATED_OPTIONS = [
+        'lint',
+        'format',
+        'skip-rule',
+        // 'extends',
+        'max-problems',
+      ];
+      const LINT_AND_BUNDLE_DOCUMENTATION_LINK =
+        'https://redocly.com/docs/cli/guides/lint-and-bundle/#lint-and-bundle-api-descriptions-with-redocly-cli';
+
+      DEPRECATED_OPTIONS.forEach((option) => {
+        if (argv[option]) {
+          process.stdout.write(
+            `Option --${option} is no longer supported. Please use separate commands, as described in the ${LINT_AND_BUNDLE_DOCUMENTATION_LINK}.`
+          );
+          process.stdout.write('\n');
+          process.exit(1);
```

Comment:
This is a great question and I don't know the right answer! I hate to break people's pipelines but we probably need to do what you've done here so they realise that action is needed on their side. Otherwise we're not linting when they think we are, and that could store up problems later.

Reply:
How about:
 - return the wording and link from the deprecation warning as an error
 - then show the usage information as we would for any other unknown command

Reply:
Looks good for me.
I have updated this part, so the output would look like =>
![Screenshot 2024-03-27 at 15 15 53](https://github.com/Redocly/redocly-cli/assets/2563120/f53f9aeb-a857-45f9-8030-46e4e3806eb8)